src/App.tsx

```diff
@@ -1,24 +1,246 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useMemo, useState } from 'react';
+import { UserWarning } from './components/UserWarning/UserWarning';
+import { client } from './utils/fetchClient';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { Errors } from './components/Errors/Errors';
+import { Status } from './types/Status';
+import { ErrorSpec } from './types/ErrorSpec';
+import { filterTodos } from './helpers/filterTodos';
+import * as TodoService from './services/todo';
 
-const USER_ID = 0;
+const USER_ID = 12021;
+const ADDED_URL = `/todos?userId=${USER_ID}`;
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [hasErrors, setHasErrors] = useState(false);
```

Comment:
it is not necessary to create a separate state to determine whether there is an error or not,
since you have an error state that stores a `ErrorSpec | null`, so we can determine whether there is an error or not by the presence of a string like this:
`const hasError = Boolean(error);`